server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
Added an issue on [girder_worker](https://github.com/girder/girder_worker/issues/46) to resolve this in a clean way.